### PR TITLE
Tutorial aircraft performance

### PR DESCRIPTION
**Pull request by pgkirsch**

### CODE REVIEW

File: aircraft/performance_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+""" A simple aircraft performance model to be used as GPkit tutorial"""
+from gpkit import Variable, Model
+from gpkit.tools import te_exp_minus1
+import numpy as np
+
+# Free Variables
+b      = Variable('b', 'm', 'Wing span')
+CD     = Variable('C_D', '-', 'Drag coefficient')
+CL     = Variable('C_L', '-', 'Lift coefficient')
+LDl    = Variable('\\frac{L}{D}_land', '-', 'Lift-to-drag in final descent')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: `'\\frac{L}{D}_{land}'` (braces on land)

File: aircraft/performance_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+M      = Variable('M', '-', 'Mach number')
+Mdd    = Variable('M_{dd}', '-', 'Drag divergence Mach number')
+R      = Variable('R', 'nautical_miles', 'Range')
+S      = Variable('S', 'm^2', 'Wing reference area')
+s_GR   = Variable('s_{GR}', 'm', 'Ground roll distance')
+s_TO   = Variable('s_{TO}', 'm', 'Total takeoff distance')
+sl     = Variable('sl', 'm', 'Total landing distance')
+sl_gr  = Variable('sl_{gr}', 'm', 'Ground roll distance (landing)')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: variable names: the difference between `s_GR` and `sl_gr` is a bit confusing / subtle

File: aircraft/performance_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+V      = Variable('V', 'knots', 'Velocity')
+Vst    = Variable('V_{stall}', 'm/s', 'Stall speed')
+W_fuel = Variable('W_{fuel}', 'lbf', 'Fuel weight')
+W      = Variable('W', 'lbf', 'Initial gross weight')
+z_bre  = Variable('z_{bre}', '-', 'Breguet parameter')
+
+# Fixed Parameters
+a     = Variable('a', 300, 'm/s', 'Speed of sound at 33,000 ft')
+AR    = Variable('AR', 11 , '-', 'Aspect Ratio')
+bmax  = Variable('b_{max}', 40, 'm', 'Gate constraint')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: is this the right value? I know they know the 737 constraint to within ~ 6" -- would be good to match

File: aircraft/performance_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+cosL  = Variable('\\cos(\\Lambda)', np.cos(25*np.pi/180), '-',
+                 'Cosine of quarter chord sweep')
+CD0   = Variable('CD0', 0.02, '-', 'Parasitic drag coefficient')
+CDl   = Variable('C_D_{land}', 0.08, '-', 'Drag coefficient in landing configuration')
+CLmaxt= Variable('C_{L_{max}}', 2.0, '-', 'Maximum lift coefficient (takeoff)')
+CLmaxl= Variable('C_{L_{max}}', 3.0, '-', 'Maximum lift coefficient (landing)')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: no need to change this for tutorial tomorrow, but feels like the takeoff and landing conditions might be natural application for VectorVariables.

File: aircraft/performance_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+e     = Variable('e', 0.8, '-', 'Oswald efficiency factor')
+h_oc  = Variable('h_{oc}', 50, 'ft', 'Obstacle clearance height (landing)')
+k     = Variable('\\kappa', 0.95, '-', 'Technology factor')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: improve descr -- Technology factor for what?

File: aircraft/performance_tutorial.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+l_rw  = Variable('l_{rw}', 5000, 'ft', 'Runway Length (DCA)')
+mu_r  = Variable('\\mu_r', 0.4, '-', 'Rolling friction coefficient')
+rho   = Variable('\\rho', 0.4, 'kg/m^3', 'Air density')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: at 33,000 ft? (make description mirror that of `a`)

File: aircraft/performance_tutorial.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+s_OC  = Variable('s_{OC}', 200, 'm', 'Obstacle clearance distance')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: takeoff or landing? Doesn't this depend on aircraft performance, given the oc height?

File: aircraft/performance_tutorial.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+# Constants
+g      = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')
+rho_sl = Variable('\\rho_{SL}', 1.225, 'kg/m^3', 'Air density at sea level')
+
+# Model
+objective = 1/R  # Maximize range
+
+constraints = [# Weight buildup
+               W >= W_e + W_pay + W_fuel,
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: make `W_e` free and constrain `W_e >= f_struct*W` (structural weight fraction)

File: aircraft/performance_tutorial.py [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+               CD >= CD0 + CL**2/(np.pi*e*AR),
+
+               # Speed limited by compressible drag rise
+               M == V/a,
+               M <= Mdd,
+               # Korn equation
+               Mdd + tau/cosL**2 + CL/(10*cosL**3) <= k/cosL, 
+
+               # Takeoff
+               s_GR >= 1.44*W**2/(g*rho_sl*S*CLmaxt*T_TO),
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: inline comment stating where 1.44 comes from

File: aircraft/performance_tutorial.py [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+CLmaxl= Variable('C_{L_{max}}', 3.0, '-', 'Maximum lift coefficient (landing)')
+e     = Variable('e', 0.8, '-', 'Oswald efficiency factor')
+h_oc  = Variable('h_{oc}', 50, 'ft', 'Obstacle clearance height (landing)')
+k     = Variable('\\kappa', 0.95, '-', 'Technology factor')
+l_rw  = Variable('l_{rw}', 5000, 'ft', 'Runway Length (DCA)')
+mu_r  = Variable('\\mu_r', 0.4, '-', 'Rolling friction coefficient')
+rho   = Variable('\\rho', 0.4, 'kg/m^3', 'Air density')
+s_OC  = Variable('s_{OC}', 200, 'm', 'Obstacle clearance distance')
+tau   = Variable('\\tau', 0.13, '-', 'Wing thickness-to-chord ratio')
+Tmax  = Variable('T_{max}', 1E6, 'N', 'Maximum thrust')
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: this is fine for now but Tmax eventually should scale with altitude

File: aircraft/performance_tutorial.py [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+               # Speed limited by compressible drag rise
+               M == V/a,
+               M <= Mdd,
+               # Korn equation
+               Mdd + tau/cosL**2 + CL/(10*cosL**3) <= k/cosL, 
+
+               # Takeoff
+               s_GR >= 1.44*W**2/(g*rho_sl*S*CLmaxt*T_TO),
+               s_TO >= s_GR +s_OC,
```
Comment (whoburg) [inline aircraft/performance_tutorial.py]: space after plus

**Comment by whoburg** [comment]
reviewed. @pgkirsch, feel free to merge after addressing comments

File: aircraft/performance_tutorial.py (hunk 1 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: Fixed

File: aircraft/performance_tutorial.py (hunk 2 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: yeah this was sloppy, renamed these all more consistently

File: aircraft/performance_tutorial.py (hunk 3 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: I believe the upper limit of category III gates is 36m. Changed to match this.

File: aircraft/performance_tutorial.py (hunk 5 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: fixed

File: aircraft/performance_tutorial.py (hunk 6 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: done

File: aircraft/performance_tutorial.py (hunk 7 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: fixed (forgot to update when I introduce landing model)

File: aircraft/performance_tutorial.py (hunk 8 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: done

File: aircraft/performance_tutorial.py (hunk 9 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: done

File: aircraft/performance_tutorial.py (hunk 11 above)
Comment (pgkirsch) [inline aircraft/performance_tutorial.py]: done

**Comment by whoburg** [comment]
:+1:

**Pull request merged**